### PR TITLE
Add Textbox.HorizontalContentAlignment and VerticalContentAlignment

### LINKED ISSUE #2634: Add Horizontal & Vertical ContentAlignment to TextBox
In Avalonia TextBox is missing properties to arrange its content
if we face this:

TextboxWithoutAlignment

In this case we need way to arrange the content to look like this:

TextboxWithAlignment

We can do it by this code:

`<TextBox Background="#eeeeee" Height="70" Width="200" Text="Some text here">
      <TextBox.Styles>
        <Style Selector="TextBox  /template/  TextPresenter">
          <Setter Property="HorizontalAlignment" Value="Center"/>
          <Setter Property="VerticalAlignment" Value="Center"/>
        </Style>
      </TextBox.Styles>
    </TextBox>`

But if we add watermark it won't get alignment, and we could fix it by this :

`<TextBox Background="#eeeeee" Height="70" Width="200" Text="Some text here">
      <TextBox.Styles>
        <Style Selector="TextBox  /template/ Panel">
          <Setter Property="HorizontalAlignment" Value="Center"/>
          <Setter Property="VerticalAlignment" Value="Center"/>
        </Style>
      </TextBox.Styles>
    </TextBox>`

But I think it's better and easier to do it WPF way like this:

` <TextBox Background="#eeeeee" Height="70" Width="200" Text="Some text here"
             VerticalContentAlignment="Center" HorizontalContentAlignment="Center">
    </TextBox>`

### PR DESCRIPTION
**Pull request by Symbai**
## What does the pull request do?
Adds two properties
```
HorizontalAlignment HorizontalContentAlignment {get; set;}
VerticalAlignment VerticalContentAlignment {get; set;}
```
 to set vertical and horizontal alignment of Textbox's content (i.e. Text and Watermark). Both properties exist in WPF but were missing in Avalonia.

## What is the current behavior?
There is no such property but developers can achieve the same by overriding styles.

## What is the updated/expected behavior with this PR?
image
```
<TextBox Text="Lorem ipsum dolor sit amet, consectetur adipiscing elit." Width="200" />
<TextBox Text="Lorem ipsum dolor sit amet, consectetur adipiscing elit." Width="400" Height="100"
		 HorizontalContentAlignment="Center" VerticalContentAlignment="Center"/>
<TextBox Watermark="ReadOnly" IsReadOnly="True" Text="This is read only"/>
<TextBox Width="200" Watermark="Watermark" />
<TextBox Width="200"
<TextBox HorizontalContentAlignment="Center" VerticalContentAlignment="Center"  Height="100" Width="200" Watermark="Watermark" />
<TextBox Width="200"  Height="100"
		 HorizontalContentAlignment="Center" VerticalContentAlignment="Center"
		 Watermark="Floating Watermark"
		 UseFloatingWatermark="True"
		 Text="Lorem ipsum dolor sit amet, consectetur adipiscing elit."/>
```
## Fixed issues
Fixes #2634

### CODE REVIEW

**Comment by kekekeks** [comment]
TextBox code is frozen till the text rework by @Gillibald is completed. We aren't touching that code except for urgent bugfixes.

**Comment by Gillibald** [comment]
Makes sense to add this. You should probably inherit the properties from here https://github.com/AvaloniaUI/Avalonia/blob/e8beffa39226cee624f45ff55f6757c9620a72e2/src/Avalonia.Controls/ContentControl.cs#L41

**Comment by Symbai** [comment]
You mean replacing textbox's TemplatedControl inherit with ContentControl? Wouldn't this bring much other properties to textbox too? Why is this better?

**Comment by Gillibald** [comment]
No. You should use `ContentControl.HorizontalContentAlignmentProperty.AddOwner` etc.

**Comment by Symbai** [comment]
Oh okay, is it better now?

File: src/Avalonia.Controls/TextBox.cs [hunk 1]
```diff
@@ -262,6 +275,17 @@ public string Text
             }
         }
 
+        public HorizontalAlignment HorizontalContentAlignment
```
Comment (MarchingCube) [inline src/Avalonia.Controls/TextBox.cs]: You could also copy documentation comment from `ContentControl` :)

File: src/Avalonia.Controls/TextBox.cs [hunk 2]
```diff
@@ -275,11 +275,17 @@ public string Text
...
 
+        /// <summary>
+        /// Gets or sets the horizontal alignment of the content within the control.
+        /// </summary>
         public HorizontalAlignment HorizontalContentAlignment
         {
             get { return GetValue(HorizontalContentAlignmentProperty); }
             set { SetValue(HorizontalContentAlignmentProperty, value); }
         }
+        /// <summary>
```
Comment (MarchingCube) [inline src/Avalonia.Controls/TextBox.cs]: Nit: New line between properties. Didn't notice this before.

**Review by MarchingCube — APPROVED** [review]
Thanks!

**Comment by FoggyFinder** [comment]
How does it affect to `TextAlignment` property? (Asking because I can't check it right now by myself)

**Comment by MarchingCube** [comment]
@FoggyFinder @Gillibald Did quick test in WPF, `HorizontalContentAlignment` is provided by `Control` so everything has it (nothing really specific to `TextBox`).

You still need `VerticalContentAlignment` if you need control over vertical layout.

[System.Windows.Controls.TextBox.TextAlignment](https://docs.microsoft.com/en-us/dotnet/api/system.windows.controls.textbox.textalignment?redirectedfrom=MSDN&view=netframework-4.8#System_Windows_Controls_TextBox_TextAlignment) documentation mentions that it takes precedence over `HorizontalContentAlignment`.

So while this PR makes our behavior a bit closer to WPF it is still not the same unfortunately. That means if you were to set `TextAlignment` to `Center` and `HorizontalContentAlignment` to `Right` text will be incorrectly positioned on the right side.

**Comment by Gillibald** [comment]
Ahh. Wasn't able to find that part. So the watermark should always follow both content alignments but the TextPresenter should only follow the vertical alignment.

For horizontal alignment we have to do the same as the Wpf implementation. 

First check if the user has set a text alignment explicitly if so apply that alignment. 

If that does not apply we look for a explicit content alignment.

After that just apply the default text alignment.

**Pull request merged**